src/main/java/com/thebois/models/beings/AbstractBeingGroup.java

```diff
@@ -8,50 +8,37 @@
  */
 public abstract class AbstractBeingGroup implements IBeingGroup {
 
-    private final Collection<IBeing> beings = new ArrayList<>();
+    private Collection<IBeing> beings = new ArrayList<>();
 
     /**
-     * Creates a BeingGroup containing 5 Beings.
-     */
-    public AbstractBeingGroup() {
-        final int numberOfBeings = 5;
-        createBeing(numberOfBeings);
-    }
-
-    /**
-     * Creates an AbstractBeingGroup containing n IBeings.
+     * Adds a being to the internal collection of beings.
      *
-     * @param numberOfBeings the number of IBeings to create.
+     * @param being IBeing to be added.
      */
-    public AbstractBeingGroup(int numberOfBeings) {
-        createBeing(numberOfBeings);
-    }
-
-    @Override
-    public Collection<IBeing> getBeings() {
-        return beings;
+    protected void addBeing(final IBeing being) {
+        beings.add(being);
     }
 
     @Override
     public void update() {
-        for (IBeing being : this.beings) {
+        for (final IBeing being : this.beings) {
             being.update();
         }
     }
 
-    /**
-     * Adds a being to the internal collection of beings.
-     *
-     * @param being IBeing to be added.
-     */
-    protected void addBeing(IBeing being) {
-        beings.add(being);
+    @Override
+    public Collection<IBeing> getBeings() {
+        final Collection<IBeing> clonedBeings = new ArrayList<>(beings.size());
+        clonedBeings.addAll(beings);
+        return clonedBeings;
```

Comment:
>Wouldn't it be possible to use Iterable here instead?

Turning my own ways against me, I see. 😆
Yeah, no:
![image](https://user-images.githubusercontent.com/12572888/134358166-1298a2aa-e6bf-4ac1-8bbb-a5b3705dcba9.png)


>And is it fine that we give a reference to the real beings and not copies of them as well? Like we do with Terrain and Structure. Since all the changes made to beings should go through BeingGroup?

Problem is that `Colony`, a subclass to `AbstractBeingGroup` needs to mutate the `IBeing`s. Since it's better to have subtypes access fields in their supertypes through a getter, `Colony` has to use `getBeings()`. Since `IBeing` represents an Entity, it should be okay since they are supposed to be mutable. There is only supposed to be a single canonical representation of a pawn in the game, for example. 

Question is, do we think `Terrain` and `Structure`s are Entities as well, or are they data objects? I'm leaning towards data objects for `Terrain` but Entity for `Structure, since terrain is just a thing while a structure is much more actively interacting with the world.